core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -476,45 +511,39 @@ DeleteFileIndex build() {
 
           List<Pair<Long, DeleteFile>> posFilesSortedBySeq =
               deleteFilesByPartition.get(partition).stream()
-                  .filter(entry -> entry.file().content() == FileContent.POSITION_DELETES)
-                  .map(entry -> Pair.of(entry.dataSequenceNumber(), entry.file()))
+                  .filter(file -> file.content() == FileContent.POSITION_DELETES)
+                  .map(file -> Pair.of(file.dataSequenceNumber(), file))
```

Comment:
is this also the entry related refactoring, maybe this could be cut into separate refactor PR and merge first ?

Reply:
Yep, I mentioned it [here](https://github.com/apache/iceberg/pull/8123#discussion_r1270132424).